README.md

```diff
@@ -34,8 +34,126 @@ While being processed, your raw excel file will be stored in `data1/excel` or `d
+
+```json
+[
+  {
+    "type": "data1",
+    "source": {
+      "headerRow": 1,
+      "startingDataRow": 2,
+      "columns": [
+        {
+          "col": "A",
+          "title": "NAME",
+          "constraints": {}
+        },
+        {
+          "col": "B",
+          "title": "DIAGNOSE"
+        }
+        // ...
+      ]
+    },
+    "destinations": [
+      {
+        "kind": "patient",
+        "columns": {
+          "inSheet": [
+            { "col": "A", "name": "name" },
+            { "col": "B", "name": "diagnose" }
+            // ...
+          ],
+          "outSheet": [
+            { "name": "status" }
+            // ...
+          ]
+        }
+      },
+      {
+        "kind": "specimen"
+        // ...
+      }
+    ]
+  },
+  {
+    "type": "data2"
+    // ...
+  }
+]
+```
+
+This app can handle multiple templates which then mapped to different databases. One entry in root array corresponds to one template.
+
+### Defining source excel file template
+
+- `headerRow`: (**required**) determines in what row is the header in excel template
+- `startingDataRow`: (**required**) determines in what row the data starts from
+- `columns`: (**required**) defines the columns of the excel template
+  - `col`: (**required**) column identifier (A, B, C, ...)
+  - `title`: (**required**) title of the column
+  - `constraints`: (**optional**) constraints of the value for the said column
+
+The constraints is using [validate.js](https://validatejs.org) syntax and you can use all standard [validators](https://validatejs.org/#validators) available.
+
+Common example of adding constraints:
+
+- The column can not be empty
+  ```json
+  {
+    "col": "A",
+    "title": "NAME",
+    "constraints": {
+      "presence": true
+    }
+  }
+  ```
+- The value should match a regex pattern
+
+  ```json
+  {
+    "col": "C",
+    "title": "GENDER",
+    "constraints": {
+      "format": {
+        "pattern": "[MF]"
+      }
+    }
+  }
+  ```
+
+- The value should exists and not shorter than a required limit
+  ```json
+  {
+    "col": "A",
+    "title": "NAME",
+    "constraints": {
+      "presence": true,
+      "length": { "minimum": 3 }
+    }
+  }
+  ```
+
+### Mapping from excel to target database table fields
+
+Data from one excel file can be inserted to multiple tables in one database, and you can pick what columns to be mapped to each table.
+
+- `kind`: (**required**) is the identifier for the table (the table name will be fetched from environment variable)
+- `columns`: (**required**)
+  - `inSheet`: (**required**) declares fields in database table that has corresponding columns in excel template
+    - `col`: (**required**) the source column in excel template (A, B, C, ...)
+    - `name`: (**required**) the target field in database table
+  - `outSheet`: declares fields in database table that does not have corresponding columns in excel template
+    - `name`: (**required**) name of field in database table
```

Comment:
Please add type to determine as a required field if the value is not varchar

Reply:
done